File: InformacjeTurystyczne/InformacjeTurystyczne/Models/Tabels/Trail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace InformacjeTurystyczne.Models.Tabels
{
    public class Trail
    {
        [Key]
        [Display(Name = "Klucz szlak")]
        public int IdTrail { get; set; } // K

        [Display(Name = "Nazwa szlaku")]
        public string Name { get; set; }

        [Display(Name = "Kolor")]
        public string Colour { get; set; }

        [Display(Name = "Otwarty")]
        public bool Open { get; set; }

        [Display(Name = "Przyczyna zamknięcia")]
        public string Feedback { get; set; }

        [Display(Name = "Długość")]
        public float Length { get; set; }

        [Display(Name = "Trudność")]
        public int Difficulty { get; set; }

        [Display(Name = "Opis")]
        public string Description { get; set; }

        public virtual ICollection<RegionLocation> RegionLocation { get; set; }
        public virtual ICollection<PermissionTrail> PermissionTrail { get; set; }
    }
}

File: InformacjeTurystyczne/InformacjeTurystyczne/wwwroot/js/trail.js
import Util from "./utility.js";

const Trail = function () {
    this.type = "szlak";
    this.name = "N/A";
    this.colour = "N/A"
    this.length = "N/A";
    this.difficulty = "N/A";
    this.open = "N/A";
    this.feedback = "N/A";
    this.description = "N/A";
    this.region = "N/A";
}

Trail.prototype.render = function () {
    let itemDiv = Util.createElement("div", { withClass: "info__item" });

    itemDiv.appendChild(Util.createElement("h1", { withText: this.name }));
    itemDiv.appendChild(Util.createElement("p", { withText: this.type, withClass: "info__type" }));
    if (Array.isArray(this.region)) {
        itemDiv.appendChild(Util.createRow("region:", this.region.join(", ")));
    } else {
        itemDiv.appendChild(Util.createRow("region:", this.region));
    }
    itemDiv.appendChild(Util.createRow("kolor:", this.colour));
    itemDiv.appendChild(Util.createRow("długość:", this.length));
    itemDiv.appendChild(Util.createRow("poziom trudności:", this.difficulty));
    itemDiv.appendChild(Util.createRow("otwarty:", this.open));
    if (this.open != "True") {
        itemDiv.appendChild(Util.createRow("przyczyna:", this.feedback));
    }
    itemDiv.appendChild(Util.createRow("opis:", this.description));

    return itemDiv;
}

export default Trail;